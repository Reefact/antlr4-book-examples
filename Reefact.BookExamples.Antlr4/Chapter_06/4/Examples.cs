#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._4 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void mermaid_style_tree() {
            // Setup
            GRun grun = GRun.ReadResource("t.cymbol", 6, 4);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree2.svg")]
        public void precedence_01() {
            // Setup
            GRun grun = GRun.ReadString("int a = -x+y;");
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree3.svg")]
        public void precedence_02() {
            // Setup
            GRun grun = GRun.ReadString("float b = -a[i];");
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

    }

}