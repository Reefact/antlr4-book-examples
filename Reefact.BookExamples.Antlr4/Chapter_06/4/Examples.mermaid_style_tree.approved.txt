graph TD
	1["file"] --> 2["varDecl"]
	2 --> 3["type"]
	3 --> 4["int"]
	2 --> 5["g"]
	2 --> 6["="]
	2 --> 7["expr"]
	7 --> 8["9"]
	2 --> 9[";"]
	1 --> 10["functionDecl"]
	10 --> 11["type"]
	11 --> 12["int"]
	10 --> 13["fact"]
	10 --> 14["("]
	10 --> 15["formalParameters"]
	15 --> 16["formalParameter"]
	16 --> 17["type"]
	17 --> 18["int"]
	16 --> 19["x"]
	10 --> 20[")"]
	10 --> 21["block"]
	21 --> 22["{"]
	21 --> 23["stat"]
	23 --> 24["if"]
	23 --> 25["expr"]
	25 --> 26["expr"]
	26 --> 27["x"]
	25 --> 28["=="]
	25 --> 29["expr"]
	29 --> 30["0"]
	23 --> 31["then"]
	23 --> 32["stat"]
	32 --> 33["return"]
	32 --> 34["expr"]
	34 --> 35["1"]
	32 --> 36[";"]
	21 --> 37["stat"]
	37 --> 38["return"]
	37 --> 39["expr"]
	39 --> 40["expr"]
	40 --> 41["x"]
	39 --> 42["*"]
	39 --> 43["expr"]
	43 --> 44["fact"]
	43 --> 45["("]
	43 --> 46["exprList"]
	46 --> 47["expr"]
	47 --> 48["expr"]
	48 --> 49["x"]
	47 --> 50["-"]
	47 --> 51["expr"]
	51 --> 52["1"]
	43 --> 53[")"]
	37 --> 54[";"]
	21 --> 55["}"]
